array/DNAm/analysis/EWAS/summariseWithinCTDMPs.r

```diff
@@ -194,47 +194,23 @@ dev.off()
 
 plotLim<-range(dmpRes[,c(4,7,10)])
 
-p1<- subset(dmpRes, DiscoveryCellType == "NeuN+") %>%
-ggplot(aes(x = `Coeff:Double-`, y = `Coeff:NeuN+`)) + geom_point() +
-xlab("Double -ve") + ylab("NeuN +ve") + ggtitle("NeuN+ve DMPs") +
-xlim(plotLim) + ylim(plotLim) +
-geom_hline(yintercept = 0) + 
-geom_vline(xintercept = 0)
-
-p2<- subset(dmpRes, DiscoveryCellType == "NeuN+") %>%
-ggplot(aes(x = `Coeff:Sox10+`, y = `Coeff:NeuN+`)) + geom_point() +
-xlab("Sox10 +ve") + ylab("NeuN +ve") + ggtitle("NeuN+ve DMPs") +
-xlim(plotLim) + ylim(plotLim) +
-geom_hline(yintercept = 0) + 
-geom_vline(xintercept = 0)
-
-p3<- subset(dmpRes, DiscoveryCellType == "Sox10+") %>%
-ggplot(aes(x = `Coeff:Double-`, y = `Coeff:Sox10+`)) + geom_point() +
-xlab("Double -ve") + ylab("Sox10 +ve") + ggtitle("Sox10+ve DMPs") +
-xlim(plotLim) + ylim(plotLim) +
-geom_hline(yintercept = 0) + 
-geom_vline(xintercept = 0)
-
-p4<- subset(dmpRes, DiscoveryCellType == "Sox10+") %>%
-ggplot(aes(x = `Coeff:NeuN+`, y = `Coeff:Sox10+`)) + geom_point() +
-xlab("NeuN +ve") + ylab("Sox10 +ve") + ggtitle("Sox10+ve DMPs") +
-xlim(plotLim) + ylim(plotLim) +
-geom_hline(yintercept = 0) + 
-geom_vline(xintercept = 0)
-
-p5<- subset(dmpRes, DiscoveryCellType == "Double-") %>%
-ggplot(aes(x = `Coeff:NeuN+`, y = `Coeff:Double-`)) + geom_point() +
-xlab("NeuN +ve") + ylab("Double -ve") + ggtitle("Double-ve DMPs") +
-xlim(plotLim) + ylim(plotLim) +
-geom_hline(yintercept = 0) + 
-geom_vline(xintercept = 0)
-
-p6<- subset(dmpRes, DiscoveryCellType == "Double-") %>%
-ggplot(aes(x = `Coeff:Sox10+`, y = `Coeff:Double-`)) + geom_point() +
-xlab("Sox10 +ve") + ylab("Double -ve") + ggtitle("Double-ve DMPs") +
-xlim(plotLim) + ylim(plotLim) +
-geom_hline(yintercept = 0) + 
-geom_vline(xintercept = 0)
+plotByStatus <- function(dmpCellType, constrastCellType, dmpRes, plotLim){
+    subset(dmpRes, DiscoveryCellType == dmpCellType) %>%
+    ggplot(aes(x = `Coeff:Double-`, y = `Coeff:NeuN+`)) + geom_point() +
+    xlab(paste0(dmpCellType, "ve")) + ylab(paste0(dmpCellType, "ve")) + ggtitle(paste0(dmpCellType, "ve DMPs")) +
+    xlim(plotLim) + ylim(plotLim) +
+    geom_hline(yintercept = 0) + 
+    geom_vline(xintercept = 0)
+
+}
+
+p1<- plotByStatus("NeuN+", "Double-", dmpRes, plotLim)
+p2<- plotByStatus("NeuN+", "Sox10+", dmpRes, plotLim)
+p3<- plotByStatus("Sox10+", "Double-", dmpRes, plotLim)
+p4<- plotByStatus("Sox10+", "NeuN+", dmpRes, plotLim)
+p5<- plotByStatus("Double-", "NeuN+", dmpRes, plotLim)
+p6<- plotByStatus("Double-", "Sox10+", dmpRes, plotLim)
```

Comment:
This is much nicer. Very clean.

Just to check, before you had:
```R
ggplot(aes(x = `Coeff:Sox10+`, y = `Coeff:NeuN+`))
```
and
```R
ggplot(aes(x = `Coeff:Double-`, y = `Coeff:NeuN+`))
```

(and many more variations on this).

But now all of the plots have:
```R
ggplot(aes(x = `Coeff:Double-`, y = `Coeff:NeuN+`))
```

I imagine this isn't what you want? I think you'll have to do some dereferencing in order to regain the original functionality.

I don't use dereferencing very much, but I believe you need to do something like:

```R
x_var <- paste0("Coeff:", contrastCellType)
y_var <- paste0("Coeff:", dmpCellType)
...
ggplot(aes(x = !!x_var, y = !!y_var)) + ...
```